nmcp/src/main/kotlin/nmcp/internal/DefaultNmcpAggregationExtension.kt

```diff
@@ -17,6 +18,11 @@ internal abstract class DefaultNmcpAggregationExtension(private val project: Pro
     override val allFiles = consumerConfiguration.incoming.artifactView { it.lenient(true) }.files
 
     override fun centralPortal(action: Action<CentralPortalOptions>) {
+        check(!centralPortalConfigured) {
+            "Nmcp: centralPortal {} must be called only once"
```

Comment:
Why is that requirement?

Reply:
I see your pain, however, the approach of "no one can override" effectively blocks overriding the value should one need it

Reply:
True, it's always a trade-off. 

To override a value, you'll have to wrap whatever convention plugin is setting it in the first place so there is ultimately a single source of truth for the value. 

If you don't own the convention plugin then you'll have to talk with the plugin maintainers so that they open an API for you. 

It's painful but so is the debugging of Gradle lifecyle. Which one is worst is a legit question that hopefully declarative Gradle will answer.